src/main/java/net/laboulangerie/laboulangeriecore/houses/nationhouse/HouseShop.java

```diff
@@ -0,0 +1,198 @@
+import net.laboulangerie.laboulangeriecore.houses.House;
+
+public class HouseShop implements Listener {
+    public static void displayShop(Player player, short page) {
+        int invSize = 27;
+        Inventory inv = Bukkit.createInventory(null, invSize, Component.text("§2Maisons de nation - page " + (page+1)));
+        List<UUID> freeHouses = LaBoulangerieCore.nationHouseHolder.getFreeHouses();
+
+        for (int i = page*(invSize-2); i < Math.min(freeHouses.size(), (page+1) * (invSize-2)); i++) {
+            UUID id = freeHouses.get(i);
+            ItemStack item = new ItemStack(Material.BRICKS);
+            ItemMeta meta = item.getItemMeta();
+            House house = LaBoulangerieCore.housesManager.getHouse(id);
+
+            meta.displayName(Component.text("§6" + house.getName()));
+            meta.lore(Arrays.asList(
+                Component.text("§bPrix : " + LaBoulangerieCore.nationHouseHolder.getHousePrice(id)),
+                Component.text(
+                    "§5Location : " + house.getAnchor().getX() + ", "
+                    + house.getAnchor().getY() + ", " + house.getAnchor().getZ()
+                )
+            ));
+
+            item.setItemMeta(meta);
+            inv.setItem(i - page*(invSize-2), item);
+        }
+        if (page > 0) {
+            ItemStack leftArrow = new ItemStack(Material.ARROW);
+            ItemMeta meta = leftArrow.getItemMeta();
+            meta.setCustomModelData(1);
+            meta.displayName(Component.text("§7Précédent"));
+            leftArrow.setItemMeta(meta);
+
+            inv.setItem((invSize-2), leftArrow);
+        }
+        if (freeHouses.size() > (page+1) * (invSize-2)) {
+            ItemStack rightArrow = new ItemStack(Material.ARROW);
+            ItemMeta meta = rightArrow.getItemMeta();
+            meta.setCustomModelData(2);
+            meta.displayName(Component.text("§7Suivant"));
+            rightArrow.setItemMeta(meta);
+
+            inv.setItem((invSize-1), rightArrow);
+        }
+        player.openInventory(inv);
+    }
+
+    @EventHandler
+    public void onClick(InventoryClickEvent event) {
+        InventoryView view = event.getView();
+        String title = PlainTextComponentSerializer.plainText().serialize(view.title());
+        if (!title.contains("Maisons de nation")) return;
+
+        event.setCancelled(true);
+
+        if (event.getClick() != ClickType.LEFT || event.getCurrentItem() == null) return;
+
+        ItemStack item = event.getCurrentItem();
+        short page = Short.parseShort(title.split("page ")[1]);
+        Player player = (Player) event.getWhoClicked();
+
+        if (item.getType() == Material.ARROW) {
+            if (item.getItemMeta().getCustomModelData() == 2) { //Right arrow is clicked
+                displayShop(player, page); //No need for page++; because page's title is aldready increased by 1 to prevent displaying "page 0"
+            }else { //Left arrow is clicked (CustomModelData = 2)
```

Comment:
CustomModelData != 2

Reply:
The comment is wrong CustomModelData equals 1 for the left arrow